api/middleware/stats.go

```diff
@@ -82,5 +85,7 @@ func RequestStats() macaron.Handler {
 }
```

Comment:
github doesn't let me comment higher up, but we can just compute the path slug once and then just pass that into the other functions